### PR TITLE
[BugFix] Fix parquet writer memory leak after exception

### PR DESCRIPTION
**Pull request by letian-jiang**
Fixes #issue

This PR can be merged after #26253 and CI thirdparty is stablized

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/test/formats/parquet/arrow_parquet_writer_test.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    arrow::Result<int64_t> Tell() const override { return arrow::Result<int64_t>(0); }
+
+    bool closed() const override { return false; }
+
+private:
+    int _counter{0};
+};
+
+TEST(ArrowParquetWriterTest, basic_test) {
```
Comment (Smith-Cruise) [inline be/test/formats/parquet/arrow_parquet_writer_test.cpp:48]: no `EXPECT_EQ(xxx)`? Just to check this ut can run normally?

Comment (letian-jiang) [reply be/test/formats/parquet/arrow_parquet_writer_test.cpp:48]: Yes. UT would crash if arrow is not fixed by https://github.com/StarRocks/starrocks/pull/26253

**Review by Smith-Cruise — APPROVED** [review]

**Review by Smith-Cruise — COMMENTED** [review]
z

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stephen-shelby — DISMISSED** [review]

**Review by trueeyu — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**